src/mono/wasm/runtime/js_support.js

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+export let DotNet = {}; // DotNet can be thought of as the exported API
+
+function configLoaded(config) {
+    const onConfigLoadedCallback = DotNet['onConfigLoaded'];
+    const onMonoRuntimeInitializedCallback = DotNet['onMonoRuntimeInitialized'];
+    // Note: onRuntimeInitialized is called by emsdk in another location
+
+    config.loaded_cb = function () {
+        // sends callback so that user knows that mono is loaded
+        onMonoRuntimeInitializedCallback(null);
+    };
+    config.fetch_file_cb = function (asset) {
+        return fetch (asset, { credentials: 'same-origin' });
+    }
+    
+    // sends callback so that user can modify the config as needed before sending it to mono but this is optional
+    // for user to handle
+    config = onConfigLoadedCallback(config) ?? config;
+
+    try
+    {
+        MONO.mono_load_runtime_and_bcl_args (config);
+    } catch (error) {
+        onMonoRuntimeInitializedCallback(error);
+    }
+}
+
+
+
+// MOSTLY COPIED FROM PR#53606 WHICH HAS NOT BEEN MERGED YET /////////////////////////////////////////////////////////
+
+// Loads the config file located in the root of the project
+// Not meant to be used outside of this class (TODO make private to this file when project converted to TS)
+function load_config() {
+    // since this file loads before emsdk we don't have environment vars yet, so we define them locally
+    const ENVIRONMENT_IS_NODE = typeof process === "object";
+    const ENVIRONMENT_IS_WEB = typeof window === "object";
+
+    if (ENVIRONMENT_IS_NODE){
+        try {
+            const config = JSON.parse(require("./mono-config.json"));
+            configLoaded(config);
+        } catch(e) {
+            configLoaded({error: "Error loading mono-config.json file from current directory"});
+        }
+    } else if (ENVIRONMENT_IS_WEB){
+        const xobj = new XMLHttpRequest();
```

Comment:
Larry made note about `fetch` polyfill.

Reply:
Yes, I am waiting to see the note, and other comments, before making any changes to make sure I understand what he means. However as this file is loaded before emscripten, I am not sure if the polyfill is ready yet.